packages/salesforcedx-vscode-core/src/commands/forceStartApexDebugLogging.ts

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  CliCommandExecutor,
+  Command,
+  CommandOutput,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import * as vscode from 'vscode';
+import { nls } from '../messages';
+import {
+  disposeTraceFlagExpiration,
+  showTraceFlagExpiration
+} from '../traceflag-time-decorator';
+import {
+  CancelResponse,
+  ContinueResponse,
+  EmptyParametersGatherer,
+  ParametersGatherer,
+  PostconditionChecker,
+  SfdxCommandlet,
+  SfdxCommandletExecutor,
+  SfdxWorkspaceChecker
+} from './commands';
+
+import { developerLogDebugLevels } from '.';
+
+export let prevApexCodeDebugLevel: string;
+export let prevVFDebugLevel: string;
+export let debugLevelId: string;
+
+const MILLISECONDS_PER_SECOND = 60000;
+const LOG_TIMER_LENGTH_MINUTES = 30;
+
+export class ForceStartApexDebugLoggingExecutor extends SfdxCommandletExecutor<
+  TraceFlagInfo
+> {
+  public build(data: TraceFlagInfo): Command {
+    return new SfdxCommandBuilder()
+      .withDescription(nls.localize('force_start_apex_debug_logging'))
+      .withArg('force:data:record:update')
+      .withFlag('--sobjecttype', 'DebugLevel')
+      .withFlag('--sobjectid', data.debugLevelId)
+      .withFlag('--values', 'ApexCode=Finest Visualforce=Finest')
+      .withArg('--usetoolingapi')
+      .build();
+  }
+
+  public execute(response: ContinueResponse<TraceFlagInfo>): void {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: vscode.workspace.rootPath
+    }).execute(cancellationToken);
+
+    this.attachExecution(execution, cancellationTokenSource, cancellationToken);
+    execution.processExitSubject.subscribe(async data => {
+      if (data != undefined && data.toString() === '0') {
+        showTraceFlagExpiration(response.data.expirationDate);
+      }
+    });
+  }
+}
+
+export type TraceFlagInfo = {
+  traceflagId: string;
+  expirationDate: string;
+  debugLevelId: string;
+};
+class TraceFlagInfoGatherer implements ParametersGatherer<TraceFlagInfo> {
+  public async gather(): Promise<
```

Comment:
The traceflag info needs to be gathered before execution of the updating the debuglevels so I put it here. Should this strictly be for _user_ input gathering?